pkg/api/pod/util.go

```diff
@@ -586,7 +586,7 @@ func dropDisabledFields(
 		})
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.ProbeTerminationGracePeriod) && !probeGracePeriodInUse(oldPodSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ProbeTerminationGracePeriod) {
```

Comment:
Will ensure this is blanked out whenever feature flag is disabled: https://github.com/kubernetes/enhancements/pull/2638/files#r632072138

Reply:
why is this being changed... this is non-standard... doesn't this break the following scenario:

1. enable feature
2. create pod with a value in the field
3. disable feature
4. try to make unrelated update to pod (update annotation or something)
5. API server drops value from the field
6. API validation fails because you changed pod spec on update?

Reply:
I hadn't thought about this breaking immutability/making it impossible to delete pods, which seems worse than "if you made a pod with the feature on and turn the feature flag off, sorry, it's still on."

I'd say our best path forward here is to keep the second condition as is, add an ACTION REQUIRED release note about the kubelet feature flag removal, and add some documentation that if a user wants to disable this on already created pods, they will have to delete and recreate the pods.

Reply:
That approach matches my expectations on the API side. I'll defer to node reviewers for kubelet feature gate discussion.